charts/nsq/templates/_helpers.tpl

```diff
@@ -76,3 +76,53 @@ nsqd selector labels
 {{- define "nsq.nsqd.selectorLabels" -}}
 app.kubernetes.io/component: nsqd
 {{- end }}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "nsq.imagePullSecrets" -}}
+  {{- $pullSecrets := list }}
+  
+  {{- range .Values.imagePullSecrets -}}
+    {{- $pullSecrets = append $pullSecrets . -}}
+  {{- end -}}
+  
+  {{- range .Values.metrics.image.pullSecrets -}}
+    {{- $pullSecrets = append $pullSecrets . -}}
+  {{- end -}}
+  {{- if (not (empty $pullSecrets)) }}
+imagePullSecrets:
+    {{- range $pullSecrets }}
+  - name: {{ . }}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+
+{{- define "nsq.serviceMonitor.namespace" -}}
+    {{- if .Values.metrics.serviceMonitor.namespace -}}
+        {{- print .Values.metrics.serviceMonitor.namespace -}}
+    {{- else -}}
+        {{- print .Release.Namespace -}}
+    {{- end }}
+{{- end -}}
+
+{{- define "nsq.prometheusRule.namespace" -}}
+    {{- if .Values.metrics.prometheusRule.namespace -}}
+        {{- print .Values.metrics.prometheusRule.namespace -}}
```

Comment:
... but this part isn't used anymore, right?

Reply:
Yeah, I left that out. next commit remove it, sorry